src/org/omegat/core/machinetranslators/YandexCloudTranslate.java

```diff
@@ -274,17 +278,19 @@ private String getIAMToken(final String oAuthToken) {
         return cachedIAMToken;
     }
 
-    private String getGlossaryConfigPart(Map<String, String> glossaryTerms) {
-        StringBuilder sb = new StringBuilder("\"glossaryConfig\":{\"glossaryData\":{\"glossaryPairs\":[");
-
-        String termsJson = glossaryTerms.entrySet().stream().limit(MAX_GLOSSARY_TERMS)
-                .map(e -> buildOneGlossaryJson(e.getKey(), e.getValue())).collect(Collectors.joining(","));
-
-        return sb.append(termsJson).append("]}},").toString();
+    private Map<String, Object> getGlossaryConfigPart(Map<String, String> glossaryTerms) {
+        long limit = MAX_GLOSSARY_TERMS;
+        Map<String, String> pairs = new TreeMap<>();
+        for (Map.Entry<String, String> e : glossaryTerms.entrySet()) {
+            if (limit-- == 0) break;
+            pairs.put("sorceText", e.getKey());
+            pairs.put("translatedText", e.getValue());
+        }
+        Map<String, Object> params = new TreeMap<>();
+        Map<String, Object> data = new TreeMap<>();
+        data.put("glossaryPairs", pairs);
+        params.put("glossaryData", data);
+        return params;
```

Comment:
I find the end to be a bit confusing. It would help if you create the maps in a consistent direction: either from root to leaf, or leaf to root.